projects/extension/src/background/index.ts

```diff
@@ -33,22 +35,61 @@ declare let window: Background
 window.manager = publicManager
 
 const l = logger("Extension")
+/**
+ * The amount of minutes that the DatabaseContentAlarm
+ * will rerun
+ */
+const periodInMinutes = 5
+
 export interface RequestRpcSend {
   method: string
   params: unknown[]
 }
 
+const setLocalStorage = async (key: string, chain: Chain) => {
+  const db = await chain.databaseContent(
+    chrome.storage.local.QUOTA_BYTES / wellKnownChains.size,
+  )
+  chrome.storage.local.set({ [key]: db })
+}
+
+const flushDatabases = (): void => {
+  for (const [key, chain] of wellKnownConnections) {
+    setLocalStorage(key, chain)
+  }
+}
+
```

Comment:
Can we please have all the code that's related in the same place, i.e:

```ts
const saveChainDbContent = async (key: string, chain: Chain) => {
  const db = await chain.databaseContent(
    chrome.storage.local.QUOTA_BYTES / wellKnownChains.size,
  )
  chrome.storage.local.set({ [key]: db })
}

const flushDatabases = (): void => {
  for (const [key, chain] of wellKnownConnections) {
    saveChainDbContent(key, chain)
  }
}

chrome.alarms.onAlarm.addListener((alarm) => {
  // make sure that the alarm needed is the one set for databaseContent retrieval
  if (alarm.name === "DatabaseContentAlarm") flushDatabases()
})
```

Also, I would inline the `fluhsDatabases` function, but I don't care that much about that.